homeassistant/components/switch/shelly.py

```diff
@@ -0,0 +1,121 @@
+"""
+Support for The Shelly Wifi switch.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.shelly/
+"""
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PATH,
+    CONF_USERNAME, CONF_PASSWORD, CONF_SWITCHES)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PATH = "/relay/0"
+
+SWITCH_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_PATH, default=DEFAULT_PATH): cv.string,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SWITCHES): vol.Schema({cv.slug: SWITCH_SCHEMA}),
+})
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Set up Shelly Wifi switches."""
+    switches = config.get('switches', {})
```

Comment:
Two items here:

1. Don't use a literal string – put this in a constant (e.g., `CONF_SWITCHES`) and utilize that.

Can this platform work if `switches` is not specified? If not, just access this directly and ideally as part of the loop:

```
for dev_name, properties in config[CONF_SWITCHES].items():
  ...
```

Reply:
resolved here:
https://github.com/ispiropoulos/home-assistant/commit/789a7ffe9ef783d8b23f8e06f80c7574cf227866